drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -180,13 +205,16 @@ static int rtc_stm32_stop(const struct device *dev)
 	return 0;
 }
 
-
-static uint32_t rtc_stm32_read(const struct device *dev)
+tick_t rtc_stm32_read(const struct device *dev)
```

Comment:
Non blocking but an additional cleanup commit which makes 2 variants of this functions depending on `COUNTER_NO_DATE` would be great for readability.
Interleaving both cases in the same function makes it unclear.

Reply:
Done.